### ISSUE TITLE
Fail2ban incorrectly blocks user access to website after starting vpn connection

### ISSUE THREAD

**Issue by micjo01** [fail2ban/fail2ban#1547]
_We will be very grateful, if your problem was described as completely as possible,
enclosing excerpts from logs (if possible within DEBUG mode, if no errors evident
within INFO mode), and configuration in particular of effected relevant settings
(e.g., with ` fail2ban-client -d | grep 'affected-jail-name' ` for a particular
jail troubleshooting).
Thank you in advance for the details, because such issues like "It does not work" 
alone could not help to resolve anything!
Thanks! (remove this paragraph and other comments upon reading)_

### Environment:

_Fill out and check (`[x]`) the boxes which apply. If your Fail2Ban version is outdated, 
and you can't verify that the issue persists in the recent release, better seek support 
from the distribution you obtained Fail2Ban from_

- Fail2Ban version (including any possible distribution suffixes): v0.91
- OS, including release name/version:  Debian 6.0.10
- [ ] Fail2Ban installed via OS/distribution mechanisms : Don't know.
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:

Our company website www.3e.eu has been running fail2ban to our satisfaction for quite a while, but recently we had complaints from users that as soon as the connect to our vpn server, the access to the company website no longer works.  When checking iptables -S I see that the public ip addresses of the remote users has indeed been added as a drop line.
As a temporary measure I've just added these ip addresses to the ignoreip in jail.conf, but since new users connect to the vpn and public ip address might change, I'm looking for a permanent solution.
Problem is that fail2ban has been installed by my predecessor, and I'm new to this software.

Can I only get basic support here, or is there also professional (paid) support that I could contact ?

_Summary here_

#### Steps to reproduce

Working from home and connecting to the company VPN server.  After that the company website (www.3e.eu) can no longer be accessed because the public ip address at our home is banned by Fail2ban.

#### Expected behavior

Fail2ban should not ban these public ip addresses.  Strangely enough this only happens after a vpn connection is established.

#### Observed behavior

Fail2ban.log registers the following line : 
2016-09-19 20:27:31,643 fail2ban.actions: WARNING [apache-overflows] Ban 109.131.153.137

From the rest of the log file I cannot determine why this is happening.

#### Any additional information

Jail.conf has the following setting for the jail 'apache-overflows' : 

[apache-overflows]

enabled = true
port    = http,https
filter  = apache-overflows
logpath = /var/log/apache*/*error.log
maxretry = 4

### Configuration, dump and another helpful excerpts

I've included the fail2ban.log and jail.conf files in attachment.

#### Any customizations done to /etc/fail2ban/ configuration

Except for the addition of a several ignoreip's I've made not customizations.
```


#### Relevant parts of /var/log/fail2ban.log file:
_preferably obtained while running fail2ban with `loglevel = 4`_

Not sure where to set loglevel=4.  Probably in jail.conf ? But since the log file show Debug entries, I assume that logging is already at an adequate level.
```
```

#### Relevant lines from monitored log files in question:

```
```
[fail2ban.log.txt](https://github.com/fail2ban/fail2ban/files/484744/fail2ban.log.txt)
[jail.conf.txt](https://github.com/fail2ban/fail2ban/files/484752/jail.conf.txt)

**Comment by yarikoptic** [comment]
Those fields in the template if the issue were for you to fill out so we don't guess on tea leaves

**Comment by micjo01** [comment]
Dear mr. Halchenko,

Can you tell me if there is commercial support available for Fail2ban ?

Thanks for providing contact details.

Best regards


Joeri Michiels

**Comment by sebres** [comment]
- the IP goes to ban if it makes `maxretry` failures within `findtime` seconds. Each failure matched a regular expression (from stock fail2ban or local customized in jail.local, etc).
- It may be, that the expression is not good enought or the matching just occurs in pre-authentication step (by handshake) and so even per success login you have one failure (in sense of your configuration of fail2ban), so normally for the "fix" it is enough to increase `maxretry` or decrease `findtime` for this jail.
- by checking of iptables, you can find the fail2ban jail name (prefixed with f2b-)
- why this IP was banned you can find in fail2ban.log (search for `[affected-jail-name] Found IP`)...
- if your fail2ban version is larger as 0.9 and database was not disabled, you can quick find there corresponding log-matches for this IP, e. g. by executing of following script:
```bash
# set your IP and db-path ...
?sudo? python -c "ip='1.2.3.4'; db='/var/lib/fail2ban/fail2ban.sqlite3';  import sys, logging; logging.basicConfig(stream=sys.stdout, level=logging.ERROR); from fail2ban.server.database import Fail2BanDb; db = Fail2BanDb(db); t = db.getBansMerged(ip=ip); print(('%d attempts, matches:\n  %s' % (t.getAttempt(), '\n  '.join(t.getMatches())) ) if t else 'NOT FOUND')"
```
- by providing fail2ban version and executing of `fail2ban-client -d | grep 'affected-jail-name'` with posting the output here, you give us more chances to follow why this (wrong) ban could occur;

BTW. At the moment we have not a commercial support (at least not central, so each developer does itself), but it will be time to think about...

**Comment by yarikoptic** [comment]
@micjo01 : @sebres gave nice howtos but again, if you just provide that starting information (version, etc) it would be MUCH EASIER to see what could have gone wrong, otherwise it is a dart in the dark.  and fail2ban.log is your big friend -- look into it ! ;)

**Comment by micjo01** [comment]
Hi,

As suggested I’ve tried increasing the maxretry value and decrease findtime, but it still made my public ip address (109.131.153.137) incorrectly appear in iptables -S.

I’ve went through fail2ban.log but couldn’t determine the reason why it is blocking my ip address.  I’ve included the file in annex.   Can you seen why this is happening ?

Kind regards

Joeri Michiels

**Comment by sebres** [comment]
> but it still made my public ip address incorrectly

- Firstly you should necessarily add your IP (or dns entry points to it) to `ignoreip` in default section of your `jail.local`, then it will be never banned. See https://github.com/fail2ban/fail2ban/blob/b2e3affaa0d3d59f6c982ab4e4052f82349aa0cb/config/jail.conf#L47
- Secondly, I suspect, that some regexp that you used may be bad in sense of possible wrong matching and could be misused from some "attackers" to ban your IP instead of real address, that made failure. 
I cannot guess which regexp, because you provide at all no info - which jail, which configuration (see above `fail2ban-client -d | grep 'affected-jail-name'`, etc.). Not even which fail2ban version you used, or whether some customization was done.

We've made the issue template not for nothing (that you've completely disregarded). (BTW I've still never seen so bad described issue like this)

> I’ve included the file in annex. Can you seen why this is happening ?

No, because we don't have any log... I suspect, that the e-mail attachments are not included if you answer to github, but you can attach files directly to the issue (drag&drop or select, see at end of comment).

**Comment by micjo01** [comment]
The template has been updated with all the details I could find.  I’ve inherited this server from my predecessor and unfortunately he left with all the technical knowledge.  So it took me a while to collect all the requested information.  Fail2ban is new to me so please forgive my lack of knowledge.

Some feedback on your previous mail :

·        Adding my IP to ignoreip helps indeed, but we have many new starters and their public ip address will have to be added every time or existing users change ip address, I can’t keep doing this for a long time.

·       I tried your suggestion ‘fail2ban-client -d | grep 'affected-jail-name' but I did not get any output.  So I added fail2ban.log on Github.  As you can see there is a ban on my public ip address 109.131.153.137 on a jail called apache-overflows.    Unfortunately I can’t understand from the log why it gets banned.

·        The version of Fail2ban is v0.91 running on a Debian 6.0.10.  I’ve also included this info in the template on Github.

Let me know if you need any other info.

Thanks
Regards


Joeri

**Comment by sebres** [comment]
> I tried your suggestion ‘fail2ban-client -d | grep 'affected-jail-name' 

The name of your affected jail is `apache-overflows`, so execute `fail2ban-client -d | grep 'apache-overflows'`

It observes `/var/log/apache/error.log`, using filter https://github.com/fail2ban/fail2ban/blob/master/config/filter.d/apache-overflows.conf... (or possibly customized in your system)
The stock fail2ban regex in this filter uses gready catch-all (`.*`), which could be theoretically "vulnerable", but stands after `<HOST>`, so IMHO it should be at least "attack-safe"...

The example of log entries, that it should ban, you can find here - https://github.com/fail2ban/fail2ban/blob/master/fail2ban/tests/files/logs/apache-overflows

The matching lines you can find using `fail2ban-regex`:
```bash
?sudo? fail2ban-regex --print-all-matched /var/log/apache/error.log /etc/fail2ban/filter.d/apache-overflows.conf
```

You can disable this jail with `enabled = false` (and restart of f2b), as long as your issue not clarified

**Comment by micjo01** [comment]
Executing  fail2ban-client -d | grep 'apache-overflows'  gives the following output (see output.txt)
[output.txt](https://github.com/fail2ban/fail2ban/files/485359/output.txt)

Executing fail2ban-regex gives me the following message :+1: 
jmi@web01:~$ sudo fail2ban-regex --print-all-matched /var/log/apache/error.log /etc/fail2ban/filter.d/apache-overflows.conf
/usr/share/fail2ban/server/filter.py:442: DeprecationWarning: the md5 module is deprecated; use hashlib instead
  import md5
Usage: /usr/bin/fail2ban-regex [OPTIONS] <LOG> <REGEX> [IGNOREREGEX]

**Comment by sebres** [comment]
> Executing ... gives the following output (see output.txt)

- Well, the failregex in given dump for this jail seems to correspond to fail2ban version fewer as 0.8.4 (too old or was customized)...
- I recommend you to reinstall fail2ban, firstly stop fail2ban, backup your configs from /etc/fail2ban, and not restore it after reinstall, but merge the custom changes only (I suspect it will be very hard to find it if they were written directly in `.conf` file).
All the changes belong to corresponding `.local` file, e. g. changes of `jail.conf` (at least `[jail] \n enabled = true` for each active jail) belong into `jail.local`, etc.

> Executing fail2ban-regex gives me the following message ... Usage: /usr/bin/fail2ban-regex ...

`--print-all-matched` should be already implemented in 0.9.1, are you sure with the fail2ban version?
What says `fail2ban-regex --help`?
